test/Spec.hs

```diff
@@ -251,6 +251,13 @@ prop_parallel prb = parallelProperty
 
 ------------------------------------------------------------------------
 
+scopeCheck :: RefKit cmd ref => [cmd ref] -> Bool
+scopeCheck = go 0
+  where
+  go _ []       = True
+  go s (c : cs) = all (\r -> r < toEnum s) (usesRefs c) &&
+    go (if returnsRef c then s + 1 else s) cs
+
```

Comment:
We could move `prop_genScope`, `prop_shrinkForkScope`, `prop_shrinkForkSubseq` to be generic test-of test functions... Actaully I see now that some of them uses `cheat` which is a bit annoying...